### ISSUE TITLE
Run tests on CI

### ISSUE THREAD

**Issue by SamVerschueren** [alexlafroscia/tldr-alfred#2]
We do it with all our workflows, it should work because `alfy-test` mocks Alfred so it doesn't need Alfred to actually being installed. What errors do you get?

Examples:
- https://github.com/SamVerschueren/alfred-ng
- https://github.com/SamVerschueren/alfred-rxjs
- https://github.com/sindresorhus/alfred-npms

**Comment by alexlafroscia** [comment]
To be honest, I didn't look into it _too_ deeply. I hit this right away, though:

https://travis-ci.org/alexlafroscia/tldr-alfred/jobs/313245680#L502

After `yarn install` it tries to run `alfy-init` and that fails.

**Comment by alexlafroscia** [comment]
So... TIL `yarn` has a flag not to run postinstall scripts! That solves that problem. I still have the problem that `nock` isn't stubbing out my call to the Github API, though, so I need to fix that for the tests to work correctly 😅 

Thanks for the examples of working repos, that was really helpful!

**Comment by alexlafroscia** [comment]
I've been unable to get Nock working... I know that you made `alfy-test` so you're the person to ask. I know that it's basically using `exec` to execute the workflow script, so stubbing things in the original JS thread won't help because it's a separate process. Any ideas on how I could stub the network requests in the test?

**Comment by SamVerschueren** [comment]
Yes, it's executed just like the Alfred workflow internally executes it with `run-node` on a different process. I actually never stubbed my http calls. I always worked with predictable data coming from the server. Is that a possibility for you?

**Comment by alexlafroscia** [comment]
Not really. In this case, we're talking to Github so I don't want to bank on that data not changing -- if someone updated the `tldr` document for the command, it would break the tests.

**Comment by alexlafroscia** [comment]
I ended up going away from `alfy-test` to just testing the workflow directly. I know that it doesn't invoke the function the same way that Alfred does, but the tradeoff is that I can actually stub the network and have reliable expectations against the responses there.

All of the logic in the workflow got moved into a `main` function that is then exported. If the `index.js` is being invoked directly (which can be detected at runtime) then the `main` function is called, passing in the input from `alfy.input` and sending the results to `alfy.output`. The `main` function itself just takes a string as input and a Promise as output, making it easy to test like any other function. By avoiding the spawning of another process, I can stub the network requests as expected.

Thanks for your help here, @SamVerschueren! I appreciate the effort even if I ended up removing `alfy-test.`
